### PR TITLE
feat: set monolog.extra

### PR DESCRIPTION
**Pull request by simPod**
https://github.com/getsentry/sentry-php/pull/1329

### CODE REVIEW

**Comment by ste93cry** [comment]
Do you mind updating the CHANGELOG and targeting the `develop` branch please?

**Comment by simPod** [comment]
@ste93cry done. Also added tests.

**Review by Jean85 — APPROVED** [review]

File: tests/Monolog/HandlerTest.php [hunk 1]
```diff
@@ -334,5 +334,53 @@ public function handleDataProvider(): iterable
...
+            false,
+            RecordFactory::create(
+                'foo bar',
+                Logger::WARNING,
+                'channel.foo',
+                [],
+                [
+                    'foo' => 'bar',
+                    'bar' => 'baz',
+                ],
```
Comment (Jean85) [inline tests/Monolog/HandlerTest.php:376]: Trailing comma isn't supported in 7.2. That's what's making the CI fail
```suggestion
                ]
```

**Review by Jean85 — CHANGES_REQUESTED** [review]

**Review by Jean85 — APPROVED** [review]

File: src/Monolog/Handler.php [hunk 2]
```diff
@@ -73,6 +73,11 @@ protected function doWrite($record): void
                 $scope->setExtra('monolog.context', $monologContextData);
             }
 
+            $monologExtraData = $this->getMonologExtraData($record['extra']);
+            if ([] !== $monologExtraData) {
```
Comment (ste93cry) [inline src/Monolog/Handler.php:77]: Any reason to not use `!empty()` as we did a few lines above? I would prefer it for consistency purposes, if not else

**Review by ste93cry — CHANGES_REQUESTED** [review]

File: src/Monolog/Handler.php (hunk 2 above)
Comment (simPod) [reply src/Monolog/Handler.php:77]: Empty is overall considered a dangerous construct, `!==[]` comparison is more strict and shows intent.

I'd rather update the empty few lines above to `[] !==` if you want consistency. WDYT?

**Comment by simPod** [comment]
@ste93cry can we expect release of this anytime soon? Thanks.

**Comment by ste93cry** [comment]
We can make a release next week. Sounds good to you?

**Comment by simPod** [comment]
Yup, thx a lot!

**Comment by ste93cry** [comment]
Released with version [`3.7.0`](https://github.com/getsentry/sentry-php/releases/tag/3.7.0) 🚢

**Pull request merged**